### PR TITLE
RAP-4279: first cut of watchserver plugin

### PR DESCRIPTION
**Pull request by jonathan-major**

### CODE REVIEW

File: Apps/WatchServerPlugin/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
+version = '3.0.4'
+
+ext {
+  workflowJarName = 'WatchServerWorkflow'
+}
+
+dependencies {
+  compile "net.rapture:SelfPluginInstallerLib:$project.version"
+  compile 'net.sourceforge.htmlunit:htmlunit:2.23'
```
Comment (dukenguyen) [inline Apps/WatchServerPlugin/build.gradle]: remove if not needed

File: Apps/WatchServerPlugin/src/main/resources/PLUGIN/content/workflows/incapture/watchserver/wsload.workflow [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                {
+                    "name": "error",
+                    "targetStep": "$FAIL"
+                }
+            ]
+        }
+    ],
+    "startStep": "setup",
+    "jarUriDependencies": [
+        "jar://workflows/incapture/wsload/*"
```
Comment (dukenguyen) [inline Apps/WatchServerPlugin/src/main/resources/PLUGIN/content/workflows/incapture/watchserver/wsload.workflow]: should correlate with line 25 of build.gradle

File: Apps/WatchServerPlugin/build.gradle (hunk 1 above)
Comment (jonathan-major) [inline Apps/WatchServerPlugin/build.gradle]: done

File: Apps/WatchServerPlugin/src/main/resources/PLUGIN/content/workflows/incapture/watchserver/wsload.workflow (hunk 2 above)
Comment (jonathan-major) [inline Apps/WatchServerPlugin/src/main/resources/PLUGIN/content/workflows/incapture/watchserver/wsload.workflow]: done

File: Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/AbstractStep.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package rapture.dp.invocable.workflow;
+
+import org.apache.log4j.Logger;
+
+import rapture.common.dp.AbstractInvocable;
+import rapture.common.util.WorkflowStepUtils;
+/**
+ * Common code for all steps goes in this class
+ */
+public abstract class AbstractStep extends AbstractInvocable<Object> {
```
Comment (scarabus) [inline Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/AbstractStep.java]: There are already three versions of AbstractStep in TradeWebPlugin, PRFSPlugin and SSSHoldingsPlugin - doesn't it make more sense to have just one version in Rapture/Libs?

Comment (jonathan-major) [inline Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/AbstractStep.java]: good point I will ticket that.

File: Apps/WatchServerPlugin/src/main/java/rapture/common/util/WorkflowStepUtils.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    public Logger getLogger() {
+        if (log == null) {
+            log = Logger.getLogger(workerUri + "/" + stepName);
+            Appender app = getAppender();
+            log.removeAppender(app);
+            log.addAppender(app);
+        }
+        return log;
```
Comment (jonathan-major) [inline Apps/WatchServerPlugin/src/main/java/rapture/common/util/WorkflowStepUtils.java]: @dukenguyen can you comment?

File: Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/StepA.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.base.Charsets;
+
+
+import rapture.common.BlobContainer;
+import rapture.common.CallingContext;
+import rapture.common.impl.jackson.JacksonUtil;
+import rapture.kernel.Kernel;
+
+public class StepA extends AbstractStep {
```
Comment (jonathan-major) [inline Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/StepA.java]: I modified the stepnames; going to add actual code next week. wanted to get the first release done.

File: Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/AbstractStep.java (hunk 3 above)
Comment (jonathan-major) [inline Apps/WatchServerPlugin/src/main/java/rapture/dp/invocable/workflow/AbstractStep.java]: @dukenguyen is working on this

**Comment by scarabus** [comment]
Merged with concerns

**Pull request merged**